lib/Monads/nondet/Nondet_Monad.thy

```diff
@@ -106,7 +106,8 @@ definition select :: "'a set \<Rightarrow> ('s,'a) nondet_monad" where
   "select A \<equiv> \<lambda>s. (A \<times> {s}, False)"
 
 definition alternative ::
-  "('s, 'a) nondet_monad \<Rightarrow> ('s, 'a) nondet_monad \<Rightarrow> ('s, 'a) nondet_monad" (infixl "\<sqinter>" 20) where
+  "('s, 'a) nondet_monad \<Rightarrow> ('s, 'a) nondet_monad \<Rightarrow> ('s, 'a) nondet_monad"
+  (infixl "\<sqinter>" 20) where
```

Comment:
There isn't an official guideline that I can see for where the infix notation should go but this is what makes sense to me to and seems to be consistent with the other examples I can find.
Any objections, and if not should I add it to the style guide?

Reply:
The infix should usually go with the type (because the type gives you the arity the infix uses). But if it would have to line break then it makes sense to me to break after the type, yes.

Reply:
Right, I'll adjust some of them to do that then, where the line breaks aren't needed.
For completeness, what about this one?
```
inductive whileLoop_terminates ::
  "('r ⇒ 's ⇒ bool) ⇒ ('r ⇒ ('s, 'r) nondet_monad) ⇒ 'r ⇒ 's ⇒ bool"
  for C B where
  ```
In my mind we would also want the line break there, since the `for C B` isn't related to the type.

Reply:
Yes, breaking before `for` makes sense.

Reply:
I agree with these outcomes.